File: javascript/leetcode/binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function(nums, target) {
    let l = 0;
    let r = nums.length;
    while (l < r) {
        m = Math.floor(l + (r - l) / 2);
        if (nums[m] > target) {
            r = m;
        } else if (nums[m] === target) {
            return m;
        } else {
            l = m + 1;
        }
    }
    return -1;
};

File: ruby/leetcode/binary-search.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer}
def search(nums, target)
    l = 0
    r = nums.size()
    while l < r
        m = l + (r - l) / 2
        if nums[m] > target
            r = m
        elsif nums[m] == target
            return m
        else
            l = m + 1
        end
    end
    return -1
end
